### PR TITLE
Move parsec to cylc.parsec

### LINKED ISSUE #2880: Move parsec to cylc package
This is an issue to discuss moving the `parsec` package to the `cylc` package. Currently Cylc contains two high level packages, `cylc` and `parsec`.

It means that installing Cylc as a Python module, with `python setup.py install` for instance, would allow users to not only do `import cylc`, but also `import parsec`.

Instead, moving it to the `cylc` package would make packaging simpler in `setup.py`, and reduce the namespaces added when installing Cylc.

### PR DESCRIPTION
**Pull request by kinow**
Close #2880 

Simply moves the old `parsec` module now to `cylc.parsec`. This will reduce the number of modules exported after the user installs cylc via pip/conda/etc.

This reduces too the risk of package collision, as there is another `parsec` package already in PYPI that could be using the `parsec` namespace already.

`pytest` and `cylc run five` passed locally. Making it a draft until functional tests have passed as well.

Cheers
Bruno

### CODE REVIEW

**Comment by kinow** [comment]
Also fixes some left overs of `setup.py`, like remove some exclusions from Codacy configuration file.

File: lib/cylc/__init__.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 import os
 import sys
-from parsec import LOG
+from cylc.parsec import LOG
```
Comment (matthewrmshin) [inline lib/cylc/__init__.py]: My horror here. You should now move the logic that defines `LOG` from `lib/cylc/parsec/__init__.py` to this module.

File: lib/cylc/cfgvalidate.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
 from isodatetime.data import Calendar, Duration, TimePoint
 from isodatetime.dumpers import TimePointDumper
 from isodatetime.parsers import DurationParser, TimePointParser
-from parsec.validate import ParsecValidator, IllegalValueError
+from cylc.parsec.validate import ParsecValidator, IllegalValueError
```
Comment (matthewrmshin) [inline lib/cylc/cfgvalidate.py]: Can now combine this module with `cylc.parsec.validate`.

Comment (kinow) [reply lib/cylc/cfgvalidate.py]: Which one should stay, and which one should go?

Comment (matthewrmshin) [reply lib/cylc/cfgvalidate.py]: `cylc.parsec.validate` can probably absorb `cylc.cfgvalidate`.

File: lib/cylc/__init__.py (hunk 1 above)
Comment (kinow) [reply lib/cylc/__init__.py]: Done, logs appear to be working as before.

File: lib/cylc/cfgvalidate.py (hunk 2 above)
Comment (kinow) [reply lib/cylc/cfgvalidate.py]: Done! Looks simpler now I think. Had to update some unit tests and a comment as well.

**Comment by kinow** [comment]
I noticed that the functional tests from `parsec` were in the wrong folder. Correcting it, they were not passing once put under the `/tests/` folder. Something with the Python code that is loaded there... so I modified the Python code a bit (just eliminated the need for `lib/python/config.py` by moving all the code to `bin/synonym.py`, and now it should - hopefully 🤞  - pass in Travis.

File: .codacy.yml [hunk 3]
```diff
@@ -3,10 +3,5 @@ exclude_paths:
   - 'doc/joss-paper/**'
   - 'etc/**'
   - 'lib/cylc/tests/**'
-  - 'lib/isodatetime/**'
-  - 'lib/jinja2/**'
-  - 'lib/markupsafe/**'
-  - 'lib/parsec/tests/**'
-  - 'licences/**'
```
Comment (kinow) [inline .codacy.yml]: This was added for parsec, but then I realized we were missing a few more lines from other folders removed recently.

**Comment by kinow** [comment]
Interesting. Promoting a draft pull request to ready-to-review does not trigger Travis. So there's only one check here.

I waited until Travis builds passed for my fork though: https://travis-ci.org/kinow/cylc-flow/builds/526686529

So it should be OK, hopefully.

**Comment by kinow** [comment]
A-ha! The old close pull request, open pull request trick. Travis build triggered!

Agent86

File: lib/cylc/__init__.py [hunk 4]
```diff
@@ -19,7 +19,12 @@
 
 import os
 import sys
-from parsec import LOG
+import logging
+
+
+LOG = logging.getLogger('cylc')  # Acceptable?
+if hasattr(logging, 'NullHandler'):  # Back compat Python <2.7
```
Comment (matthewrmshin) [inline lib/cylc/__init__.py]: Can remove the back compat `if ...` test and just add the `NullHandler`.

Comment (kinow) [reply lib/cylc/__init__.py]: Sure thing! Done in 2a405368caf553563c0256d4c8552622ab6e870a

**Comment by matthewrmshin** [comment]
Sorry, but we should still add the `LOG.addHandler(logging.NullHandler())` statement, just without the need for the `if ...` test.

**Comment by kinow** [comment]
Oh, that line had to stay? No problemo. Done in cc6ed7d2b6f9f5cf7727e4ddc06cf0fa56b6f452

**Comment by matthewrmshin** [comment]
Will merge when CI is done.

**Pull request merged**